### PR TITLE
[libcint] Update to version 5.1.1

### PR DESCRIPTION
**Pull request by panxl**
Also removed the dependency on OpenBLAS since the library doesn't depend on BLAS anymore since v5.0.0.

### CODE REVIEW

File: L/libcint/build_tarballs.jl [hunk 1]
```diff
@@ -39,7 +38,6 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
```
Comment (giordano) [inline L/libcint/build_tarballs.jl:40]:
```suggestion
dependencies = Dependency[
```

**Review by giordano — COMMENTED** [review]
@panxl can we require Julia v1.6?

File: L/libcint/build_tarballs.jl (hunk 1 above)
Comment (panxl) [reply L/libcint/build_tarballs.jl]: Thanks for the fix!

**Comment by panxl** [comment]
@giordano I think it is reasonable. Is this the right way to add it?

**Comment by giordano** [comment]
Thanks!

**Pull request merged**